actionmailer/lib/action_mailer.rb

```diff
@@ -52,6 +52,7 @@ module ActionMailer
   autoload :TestHelper
   autoload :MessageDelivery
   autoload :DeliveryJob
+  autoload :NewDeliveryJob
```

Comment:
I'm not sure `NewDeliveryJob` is the best name. We don't want users to have to change the constant name back after we fully remove `DeliveryJob` (and after a few years `NewDeliveryJob` will be a strange name cause it's not new anymore). What about a name that better describes the roll of this job AND has the new code like `MailDeliveryJob`?

Reply:
My plan was to deprecate `NewDeliveryJob` and rename to `DeliveryJob` in `6.2` or `7.0` but that will also cause problems for people that upgrade multiple versions at a time. I'll rename to `MailDeliveryJob`!